nodewatcher/modules/devices/xiaomi/mi_router.py

```diff
@@ -0,0 +1,174 @@
+    url = 'http://miwifi.com/'
+    architecture = 'ramips_mt7620'
+    radios = [
+        cgm_devices.IntegratedRadio('wifi0', _("Integrated wireless radio (5 GHz)"), [
+            cgm_protocols.IEEE80211AC(
+                cgm_protocols.IEEE80211AC.SHORT_GI_20,
+                cgm_protocols.IEEE80211AC.SHORT_GI_40,
+                cgm_protocols.IEEE80211AC.RX_STBC1,
+                cgm_protocols.IEEE80211AC.DSSS_CCK_40,
+            )
+        ], [
+            cgm_devices.AntennaConnector('a1', "Antenna1")
+        ], [
+            cgm_devices.DeviceRadio.MultipleSSID,
+        ]),
+        cgm_devices.IntegratedRadio('wifi1', _("Integrated wireless radio (2.4 GHz)"), [
+            cgm_protocols.IEEE80211BGN(
+                cgm_protocols.IEEE80211BGN.SHORT_GI_20,
+                cgm_protocols.IEEE80211BGN.SHORT_GI_40,
+                cgm_protocols.IEEE80211BGN.RX_STBC1,
+                cgm_protocols.IEEE80211BGN.DSSS_CCK_40,
+            )
+        ], [
+            cgm_devices.AntennaConnector('a2', "Antenna2")
+        ], [
+            cgm_devices.DeviceRadio.MultipleSSID,
+        ])
+    ]
+    switches = [
+        cgm_devices.Switch(
+            'sw0', "Switch0",
+            ports=[0, 1, 2, 3, 4, 6],
+            cpu_port=6,
+            cpu_tagged=True,
+            vlans=16,
+            configurable=True,
+            presets=[
+                cgm_devices.SwitchPreset('default', _("Default VLAN configuration"), vlans=[
+                    cgm_devices.SwitchVLANPreset(
+                        'wan0', "Wan0",
+                        vlan=2,
+                        ports=[4, 6],
+                    ),
+                    cgm_devices.SwitchVLANPreset(
+                        'lan0', "Lan0",
+                        vlan=1,
+                        ports=[0, 1, 2, 3, 6],
+                    ),
+                ])
+            ]
+        ),
+    ]
+    ports = []
+    antennas = [
+        # TODO: This information is probably not correct
+        cgm_devices.InternalAntenna(
+            identifier='a1',
+            polarization='dual',
+            angle_horizontal=360,
+            angle_vertical=360,
+            gain=3,
+        ),
+        cgm_devices.InternalAntenna(
+            identifier='a2',
+            polarization='dual',
+            angle_horizontal=360,
+            angle_vertical=360,
+            gain=3,
+        )
+    ]
+    port_map = {
+        'openwrt': {
+            'wifi0': 'radio0',
+            'wifi1': 'radio1',
+            'sw0': cgm_devices.SwitchPortMap('switch0', vlans='eth0.{vlan}'),
+        }
+    }
+    drivers = {
+        'openwrt': {
+            'wifi0': 'mac80211',
+            'wifi1': 'mac80211',
+        }
+    }
+    profiles = {
+        'lede': {
+            'name': 'miwifi-mini',
+            'files': [
+                '*-mt7620-generic-miwifi-mini-squashfs-sysupgrade.bin',
+            ]
+        }
+    }
+
+
+class XiaomiMiNano(XiaomiMiMini):
```

Comment:
Since it is a completely different architecture I think it would be better not to reuse anything and just copy the switch over (and not extend `XiaomiMiMini`).